gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -427,6 +476,48 @@ public void delete(URI path, boolean recursive) throws IOException {
     repairImplicitDirectory(parentInfoFuture);
   }
 
+  private String getBucketName(URI path) {
+    if (path.toString() == null || path.toString() == "") return "";
+    int ind = path.toString().replace("gs://", "").indexOf("/");
+    return path.toString().replace("gs://", "").substring(0, ind);
+  }
+
+  private String getFolderName(URI path) {
+    if (path.getPath() == null || path.getPath() == "") return "";
+    return path.getPath().substring(path.getPath().indexOf("/") + 1);
+  }
+
+  /**
+   * Returns the list of folder resources in the prefix. It lists either all the folder resources or
+   * lists at max of 1 folder resource other than prefix
+   *
+   * @param prefix the prefix to use to list all matching folder resources.
+   * @param pageToken the page token to list
+   * @param listFolderOptions the page token to list
+   */
+  public ListPage<FolderInfo> listFoldersInfoForPrefixPage(
+      URI prefix, ListFolderOptions listFolderOptions, String pageToken) throws IOException {
+    logger.atFiner().log(
+        "listFoldersInfoForPrefixPage(prefix: %s, pageToken:%s)", prefix, pageToken);
+    StorageResourceId prefixId = getPrefixId(prefix);
+    ListPage<FolderInfo> itemInfosPage =
```

Comment:
nit: inline this?

Reply:
Please explain inline what ?

Reply:
i.e. just return from here, instead of having itemInfosPage variable.